nzgmdb/management/config.py

```diff
@@ -2,11 +2,41 @@
 Module to manage the configuration file for constants and configuration settings for an NZGMDB run.
 """
 
+from enum import Enum
+
 import yaml
 
 from nzgmdb.management import file_structure
 
 
+class MachineName(str, Enum):
+    """
+    Enum for the machine names.
+    """
+
+    LOCAL = "local"
+    MANTLE = "mantle"
+    HYPOCENTRE = "hypocentre"
+
+
+class WorkflowStep(str, Enum):
+    """
+    Enum for the workflow steps.
+    """
+
+    GEONET = "geonet"
+    TEC_DOMAIN = "tec_domain"
+    PHASE_TABLE = "phase_table"
+    SNR = "snr"
+    FMAX = "fmax"
+    GMC = "gmc"
+    PROCESS = "process"
+    IM = "im"
+    DISTANCES = "distances"
+    UPLOAD = "upload"
+    DEFAULT = "default"
```

Comment:
You should use `StrEnum` which is the same thing as inheriting from `str` and `Enum` but built-in.

```suggestion
from enum import StrEnum

import yaml

from nzgmdb.management import file_structure


class MachineName(StrEnum):
    """
    Enum for the machine names.
    """

    LOCAL = "local"
    MANTLE = "mantle"
    HYPOCENTRE = "hypocentre"


class WorkflowStep(StrEnum):
    """
    Enum for the workflow steps.
    """

    GEONET = "geonet"
    TEC_DOMAIN = "tec_domain"
    PHASE_TABLE = "phase_table"
    SNR = "snr"
    FMAX = "fmax"
    GMC = "gmc"
    PROCESS = "process"
    IM = "im"
    DISTANCES = "distances"
    UPLOAD = "upload"
    DEFAULT = "default"
```